### PR TITLE
add an option to specify a list of users whose commits do not trigger the hook

### PR DESCRIPTION
**Pull request by marbon87**

### CODE REVIEW

**Comment by enriquecaballero** [comment]
+1

**Comment by Steinsen** [comment]
👍

**Comment by dantran** [comment]
what is the use case for this?  for repo admin?

**Comment by Steinsen** [comment]
@dantran in my case I want to prevent loop occuring when I perform automatic release (and checkin of pom files) on a release branch.

**Comment by dantran** [comment]
ah, makes sense

Thanks

-D

File: src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java [hunk 1]
```diff
@@ -137,6 +137,25 @@ private void sendEvents(Object event, Repository repo, EventType eventType) thro
         StringEntity bodyEntity = new StringEntity(jsonBody, ContentType.APPLICATION_JSON);
         for (WebHookConfiguration webHookConfiguration : webHookConfigurationDao.getEnabledWebHookConfigurations(repo))
         {
+            if(event instanceof BitbucketPushEvent)
+            {
+                BitbucketPushEvent bitbucketPushEvent = (BitbucketPushEvent) event;
+                if (webHookConfiguration.getCommittersToIgnore() != null &&
+                        bitbucketPushEvent.getActor() != null &&
+                        webHookConfiguration.getCommittersToIgnore().contains(bitbucketPushEvent.getActor().getUsername()))
```
Comment (casz) [inline src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java]: This does not seem safe, when user names could arbitrarily contain multiple users for in my company it's based on initials. JOM and JOMR.
So JOM would match both.

**Comment by marbon87** [comment]
Good point. I fixed that

**Comment by casz** [comment]
@marbon87 could you fix your last commit. it seems to be missing user.email 👍

**Comment by marbon87** [comment]
commits squashed

**Comment by marbon87** [comment]
Any update on this? can the PR be merged?

**Comment by marbon87** [comment]
i merged the current master. Any possiblity to merge and releases this soon?

**Comment by casz** [comment]
I would prefer to get #57 and #61 in first, before merging this in.

**Comment by marbon87** [comment]
ok, please give me a message when you ready and something needed to be merged

**Comment by casz** [comment]
I am fixing merge conflict  👍

**Comment by casz** [comment]
@marbon87 In the merge, I also fixed annotation of `@Accessor` `@Mutator` as the were incorrectly placed. Accessor was on setter, and Mutator was on getter 🍰

**Comment by casz** [comment]
Fixed actual bug in code, due to new tests 💜 @bturner

File: src/main/java/nl/topicus/bitbucket/api/WebHookConfigurationModel.java [hunk 2]
```diff
@@ -14,12 +14,15 @@
...
     @XmlElement
+    private String committersToIgnore;
+    @XmlElement
     private boolean enabled;
 
     WebHookConfigurationModel(WebHookConfiguration webHookConfiguration) {
         id = webHookConfiguration.getID();
         title = webHookConfiguration.getTitle();
         url = webHookConfiguration.getURL();
+        committersToIgnore = webHookConfiguration.getCommittersToIgnore();
```
Comment (casz) [inline src/main/java/nl/topicus/bitbucket/api/WebHookConfigurationModel.java]: Was missing previously.

**Comment by casz** [comment]
@marbon87 Does the PR look good to you, now that I have rebased and squashed and fixed the tests/build? 🚀

**Comment by marbon87** [comment]
Looks good to me

**Comment by marbon87** [comment]
Great! Any plans for a new release?

**Comment by casz** [comment]
@marbon87 I'd like to test on our servers before a release. We had some great contribution with a lot of changes. 👍

**Comment by casz** [comment]
@marbon87 Release has been cut! https://marketplace.atlassian.com/plugins/nl.topicus.bitbucket.bitbucket-webhooks/versions#b100500000

**Pull request merged**